### ISSUE TITLE
GroupLimit create an error

### ISSUE THREAD

**Issue by Barbapapazes** [lookinlab/adonis-lucid-soft-deletes#8]
Hello,

Using Soft deletes on a preloaded model with [group limit](https://docs.adonisjs.com/reference/orm/relations/has-many-through#grouplimit) will create an error : 

```ts
 const associations = await Association.query()
      .preload('tags', (tags) => tags.groupLimit(3))
      .paginate(page, this.PER_PAGE)
```

There is an error because the no retrieval of deleted items clause is added at the end of the query.

![image](https://user-images.githubusercontent.com/45267552/154030766-7b90fc4f-1aeb-494c-8cf5-8de042c248e0.png)

**Comment by LookinGit** [comment]
Hi, @Barbapapazes
If you disable Soft Deletes for tags, does the query work?

**Comment by Barbapapazes** [comment]
Yes ! It works fine without SoftDeletes

**Comment by LookinGit** [comment]
Fixed in v1.4.1
